content/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/setting-up-your-php-project-for-codespaces.md

```diff
@@ -0,0 +1,187 @@
+     // Configure properties specific to VS Code.
+     "vscode": {
+       "extensions": [
+         "ikappas.composer"
+       ]
+     }
+   },
+   ```
+
+1. Add a comma after the `features` property.
+
+   ```json
+   "features": {
+     "ghcr.io/devcontainers/features/github-cli:1": {}
+   },
+   ```
+
+1. Uncomment the `postCreateCommand` property and change its value to run the command `composer install` if the `composer.json` file exists.
+
+   ```json copy
+   // Use 'postCreateCommand' to run commands after the container is created.
+   "postCreateCommand": "if [ -f composer.json ];then composer install;fi"
+   ```
+
+  The `devcontainer.json` file should now look similar to this, depending on which image you chose:
+
+   ```json
+   // For format details, see https://aka.ms/devcontainer.json. For config options, see the
+   // README at: https://github.com/devcontainers/templates/tree/main/src/php
+   {
+     "name": "PHP",
+     // Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+     "image": "mcr.microsoft.com/devcontainers/php:1-8.2-bullseye",
+
+     // Configure tool-specific properties.
+     "customizations": {
+       // Configure properties specific to VS Code.
+       "vscode": {
+         "extensions": [
+           "ikappas.composer"
+         ]
+       }
+     },
+
+     // Use 'forwardPorts' to make a list of ports inside the container available locally.
+     "forwardPorts": [
+       8080
+     ],
+     "features": {
+       "ghcr.io/devcontainers/features/github-cli:1": {}
+     },
+
+     // Use 'postCreateCommand' to run commands after the container is created.
+     "postCreateCommand": "if [ -f composer.json ];then composer install;fi"
+
+     // Uncomment to connect as root instead. More info: https://aka.ms/dev-containers-non-root.
+     // "remoteUser": "root"
+   }
+   ```
+
+{% data reusables.codespaces.save-changes %}
+{% data reusables.codespaces.rebuild-command %}
+{% indented_data_reference reusables.codespaces.rebuild-reason %}
+
+   After the dev container is rebuilt, and your codespace becomes available again, the `postCreateCommand` will have been run, installing your Composer dependencies, and the "Composer" extension will be available for use.
+
+## Step 4: Run your application
+
+In the previous section, you used the `postCreateCommand` to install a set of packages via the `composer install` command. With the dependencies now installed, you can run the application. However, in this scenario we first need to tweak the ports that Apache will listen on. By default, it is configured to use port 80, but in Codespaces, that port is not available -- so we will use port 8080 instead.
+
+1. In the Terminal of your codespace, enter `sudo sed -i 's/Listen 80$//' /etc/apache2/ports.conf`.
+
+1. Still in the Terminal, enter `sudo sed -i 's/80/8080/' /etc/apache2/sites-enabled/000-default.conf`.
+
+1. And finally, in the Terminal, enter `apache2ctl start`.
+
```

Comment:
When I run `apache2ctl start` I get:

```
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 127.0.0.1. Set the 'ServerName' directive globally to suppress this message
```

We should either mention this, or add something to prevent it being displayed.